src/java/main/org/apache/zookeeper/client/StaticHostProvider.java

```diff
@@ -113,5 +124,21 @@ public InetSocketAddress next(long spinDelay) {
 
     public void onConnected() {
         lastIndex = currentIndex;
+        connectedSinceNext = true;
+    }
+
+    private List<InetSocketAddress> resolveAddress(InetSocketAddress address) throws UnknownHostException {
+        InetAddress resolvedAddresses[] = InetAddress.getAllByName((address.getHostString()));
```

Comment:
The previous implementation checked for `address != null`. Is that no longer valid here? It seems as if the additional logic added to `next()` covers the else case of the ternary operator, but we may still want to check for `null` here to prevent an NPE.

nit: too many parentheses.

Reply:
For an NPE I think it would require someone injecting a null into the initial list.

Reply:
The constructor `StaticHostProvider` accepts a `Collection` of `InetSocketAddress`es which could contain `null` elements. If you think it's a non-issue though, I'm happy to leave it as is.

Reply:
Added an exception